tests/basic_test.go

```diff
@@ -60,6 +60,23 @@ func establishConnection(t *testing.T) scp.Client {
 	return client
 }
 
+func establishPreserveConnection(t *testing.T) scp.Client {
```

Comment:
This seems to introduce a lot of code duplication for simply passing a `true` to the `Preserve` fujnction, perhaps just add a boolean to the original function and set it to false in all other test cases.